### PR TITLE
Add validate spec command

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description

Adds a command that we could run in CI to validate the schema. This is just a rough proposal. I see that it is not catching the lack of trailing slash on the `healthcheck` endpoint for example. Maybe there's some configuration we need to do with the `validate_spec`, like maybe pass a specific spec_url to use.

## Testing Instructions

Run `just build` so that you get the new dependencies. Then run `just dj validateopenapischema` and verify that it does not fail with validation errors.

If you want to see it working (i.e., failing) you can undo some of the changes I made like removing the trailing slashes from some URL configurations or adding back that `unique=True` on the auth_token endpoint configuration. The latter yields a very mysterious error that took me quite a while to find a solution for! This webapp gave me a clear, descriptive error that was much more helpful than the validate_spec error: https://editor.swagger.io/

Maybe there's a way to use the validation underlying that in the future as an iterative improvement on this.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
I'm glad this is so easy! I don't have any advice for the non-OpenAPI-spec URLs unfortunately, but I would love to add this as part of CI too like you mention 🚀

**Review by krysal — COMMENTED** [review]
Many packages are being updated here, do you think [these instructions](https://github.com/pypa/pipenv/issues/2665#issuecomment-524919810) could work for installing the new ones? 🤔

File: api/catalog/management/commands/validateopenapischema.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def handle(self, *args, **options):
+        file_path = str(
+            (Path(options["output_dir"]).absolute() / "openapi.yaml").absolute()
+        )
+        call_command("generate_swagger", file_path, overwrite=True)
+        spec_dict, spec_url = read_from_filename(file_path)
+        # Do not catch the error here so that we can still write the file
+        # out for debugging.
+        self.handle_validation(spec_dict)
+        os.remove(file_path)
```
Comment (krysal) [inline api/catalog/management/commands/validateopenapischema.py:45]: Using [`Path.unlink()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink) could save us the `os` import and convert the path to a string.

**Review by krysal — APPROVED** [review]
This is very helpful, thanks! I left two suggestions but this looks already g2g. Sorry for the late review.

File: api/catalog/api/views/oauth2_views.py [hunk 2]
```diff
@@ -80,7 +80,6 @@ class Register(APIView):
                 type=openapi.TYPE_STRING,
                 min_length=1,
                 max_length=150,
-                unique=True,
                 description="A unique human-readable name for your application "
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply api/catalog/api/views/oauth2_views.py:83]: It does actually still have to be unique on the model side, it's just that the `unique` keyword argument on this function doesn't exist:

https://github.com/WordPress/openverse-api/blob/419082f155372ee05bdc5c4261feea06906eb49f/api/catalog/api/models/oauth.py#L11-L16

[the remark this answers is not in the document]
Comment (krysal) [reply api/catalog/api/views/oauth2_views.py:83]: Ah, nice! I thought it was a bit weird it had to be unique among **all** apps (uniqueness with the email would make more sense), but I guess that makes our management job easier 🤷🏻‍♀️

File: q [hunk 3]
```diff
@@ -0,0 +1,59 @@
+[1mdiff --cc .github/workflows/ci_cd.yml[m
+[1mindex 4c936852,1f70810a..00000000[m
+[1m--- a/.github/workflows/ci_cd.yml[m
```
Comment (krysal) [inline q:3]: What is this file? 😳

Comment (sarayourfriend) [reply q:3]: Oh gosh, I must have somehow accidentally piped `git diff` output to a file during the rebase :facepalm: It was a messy rebase :sweat_smile:

**Pull request merged**